### PR TITLE
Transition to api v2

### PR DESCRIPTION
**Pull request by Gauravjeetsingh**
* [x] Gets hukamnama from v2 api; 
* [x] gets shabad from v2
* [x] test sttm with newer API

### CODE REVIEW

**Comment by bogas04** [comment]
I've a suggestion, can we have `api/index.ts` file that does all the API fetching logic as well as accessing logic. Like instead of having `shabad.a.b.c` in our components we do `getC(shabad)` in our components and the logic sits in a helper file, making us more resilient to API breakages.

**Comment by Gauravjeetsingh** [comment]
@bogas04 ji, Yeah, this is not scalable. Initially, I did the API changes manually in code to have working code with v2 API and hukamnama fix.

As making it scalable requires more time and effort we can work on it behind the scenes, without delaying the fix to production. 

p.s. this PR is still WIP, currently testing it out, to make sure that the transition of v1 to v2 API didn't break anything.

**Comment by Gauravjeetsingh** [comment]
@bogas04 ji, I was thinking of working on scaling it, afterwards we fix this issue. Thoughts?

In this pull request, I am changing the api url in sttm-web, will remove it once we have updated version of banidb package

File: src/js/pages/Ang/index.js [hunk 1]
```diff
@@ -13,7 +13,8 @@ export default class Ang extends React.PureComponent {
   };
   render() {
     const { ang, source, highlight } = this.props;
-    const url = buildApiUrl({ ang, source, API_URL });
+    let url = buildApiUrl({ ang, source, API_URL });
+    url = url.replace('ang', 'angs');
```
Comment (bogas04) [inline src/js/pages/Ang/index.js]: We can update `buildApiUrl` and bump up the version there. https://github.com/KhalisFoundation/banidb

File: src/js/pages/Hukamnama/Layout.js [hunk 2]
```diff
@@ -13,16 +13,27 @@ export default class Layout extends React.PureComponent {
...
   render() {
     const { data } = this.props;
+    const { shabads } = data;
+    const { date, month, year } = data.date.gregorian;
+    const dateString = date + '-' + month + '-' + year;
+    let totalVerses = [];
+
+    shabads.forEach(s => {
+      totalVerses = totalVerses.concat(s.verses);
+    });
```
Comment (bogas04) [inline src/js/pages/Hukamnama/Layout.js]: `const totalVerses = shabads.map(s => s.verses)` ?

File: src/js/pages/Hukamnama/index.js [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
 import React from 'react';
-import { buildApiUrl } from '@sttm/banidb';
+//import { buildApiUrl } from '@sttm/banidb';
 import PageLoader from '../PageLoader';
 import Layout, { Stub } from './Layout';
 
 export default class Hukamnama extends React.PureComponent {
   render() {
-    const url = buildApiUrl({ hukam: true, API_URL });
+    const url = `${API_URL}hukamnamas`;
```
Comment (bogas04) [inline src/js/pages/Hukamnama/index.js]: We can update banidb package.

File: src/js/pages/Search/index.js [hunk 4]
```diff
@@ -39,21 +39,23 @@ export default class Search extends React.PureComponent {
...
                 (_, i) => i + 1
               )}
-              totalResults={pageinfo.totalresults || 0}
-              resultsCount={pageinfo.pageresults || 0}
+              totalResults={resultsInfo.totalResults || 0}
+              resultsCount={resultsInfo.pageResults || 0}
               offset={offset}
-              nextPageOffset={pageinfo.nextpageoffset}
-              shabads={shabads}
+              //nextPageOffset={resultsInfo.pages.page}
```
Comment (bogas04) [inline src/js/pages/Search/index.js]: This isn't needed anymore?

File: src/js/pages/Shabad/index.js [hunk 5]
```diff
@@ -20,9 +20,10 @@ export default class Shabad extends React.PureComponent {
 
   render() {
     const { random, id, highlight } = this.props;
-    const url = buildApiUrl(
+    let url = buildApiUrl(
       random ? { random, API_URL } : { random, id, API_URL }
     );
+    url = url.replace('shabad', 'shabads');
```
Comment (bogas04) [inline src/js/pages/Shabad/index.js]: We can update banidb package.

File: src/js/pages/Hukamnama/Layout.js (hunk 2 above)
Comment (Gauravjeetsingh) [reply src/js/pages/Hukamnama/Layout.js]: This creates a separate array for verses of each Shabad.
With concat, we get one array with all verses.

File: src/js/pages/Hukamnama/index.js (hunk 3 above)
Comment (Gauravjeetsingh) [reply src/js/pages/Hukamnama/index.js]: Bumped the version of banidb in my existing PR.

File: src/js/pages/Search/index.js (hunk 4 above)
Comment (Gauravjeetsingh) [reply src/js/pages/Search/index.js]: Yeah, forgot to ask, this wasn't used anywhere. Looked like some old code, we forgot to remove.
And, also it's not in the response of v2 API.

File: src/js/pages/Ang/index.js [hunk 6]
```diff
@@ -13,7 +13,7 @@ export default class Ang extends React.PureComponent {
   };
   render() {
     const { ang, source, highlight } = this.props;
-    const url = buildApiUrl({ ang, source, API_URL });
+    let url = buildApiUrl({ ang, source, API_URL });
```
Comment (bogas04) [inline src/js/pages/Ang/index.js]: Any reason for this?

File: src/js/pages/Hukamnama/index.js (hunk 3 above)
Comment (bogas04) [reply src/js/pages/Hukamnama/index.js]: We want to use buildApiUrl here right?

File: src/js/pages/Shabad/index.js [hunk 7]
```diff
@@ -20,7 +20,7 @@ export default class Shabad extends React.PureComponent {
 
   render() {
     const { random, id, highlight } = this.props;
-    const url = buildApiUrl(
+    let url = buildApiUrl(
```
Comment (bogas04) [inline src/js/pages/Shabad/index.js]: Why let?

File: src/js/components/Breadcrumb/Breadcrumb.js [hunk 8]
```diff
@@ -35,7 +36,7 @@ export default class BreadCrumb extends React.PureComponent {
             ) : (
               <span className="breadcrumbLast">{title}</span>
             )
-        )}
+          )}
       </h4>
     );
   }
```
[the remark this answers is not in the document]
Comment (Gauravjeetsingh) [reply src/js/components/Breadcrumb/Breadcrumb.js]: yeah, just the indentation.

**Comment by Gauravjeetsingh** [comment]
Merging to dev.

**Pull request merged**